File: app/CausalityCSharp/CausalityLibrary/Type/Context.cs
using System;

namespace CausalityLibrary.Type
{
    [Serializable]
    public sealed class Context
    {
        public Guid GUID
        { get; }

        public SerialNumber CurrentScenario
        { get; set; }

        public SerialNumber CurrentScene
        { get; set; }

        public SerialNumber CurrentCaption
        { get; set; }

        public Context(Guid guid)
        {
            GUID = guid;
        }
    }
}

File: app/CausalityJava/src/com/inha/dsp/causality/type/Context.java
package com.inha.dsp.causality.type;

import java.io.Serializable;
import java.util.UUID;

public class Context implements Serializable {
    private UUID _guid;
    public UUID getGUID() {
        return _guid;
    }
    public SerialNumber CurrentScenario;
    public SerialNumber CurrentScene;
    public SerialNumber CurrentCaption;

    public Context(UUID guid)
    {
        _guid = guid;
    }
}
